src/views/home/components/token-list/token-list.view.tsx

```diff
@@ -1,287 +1,269 @@
+          autoFocus
+          placeholder="Enter token name or address"
+          type="search"
+          className={classes.searchInput}
+          value={searchInputValue}
+          onChange={(event) => {
+            handleSearchInputchange(event.target.value);
+          }}
+        />
+        {searchInputValue !== "" && (
+          <button className={classes.clearSearchButton} onClick={() => handleSearchInputchange("")}>
+            <XMarkIcon className={classes.clearSearchButtonIcon} />
+          </button>
+        )}
+      </div>
+      <div className={classes.list}>
+        {customToken.status === "loading" ? (
+          <div className={classes.centeredElement}>
+            <Spinner size={24} />
+          </div>
+        ) : error ? (
+          <Typography type="body2" className={classes.centeredElement}>
+            {error}
+          </Typography>
+        ) : (
+          filteredTokens.map((token) => {
+            const isImportedCustomToken = isChainCustomToken(token, chains.from);
+            const isNonImportedCustomToken =
+              !isImportedCustomToken &&
+              customToken.status === "successful" &&
+              customToken.data.address === token.address;
+            return (
+              <div key={token.address} className={classes.tokenButtonWrapper}>
+                <button
+                  role="button"
+                  className={classes.tokenButton}
+                  onClick={() => onSelectToken(token)}
+                >
+                  <div className={classes.tokenInfo}>
+                    <Icon url={token.logoURI} size={24} className={classes.tokenIcon} />
+                    <Typography type="body1">{token.name}</Typography>
+                  </div>
                 </button>
-              </div>
-              <div className={classes.searchInputContainer}>
-                <MagnifyingGlassIcon className={classes.searchIcon} />
-                <input
-                  ref={inputRef}
-                  placeholder="Enter token name or address"
-                  type="search"
-                  className={classes.searchInput}
-                  value={searchInputValue}
-                  onChange={(event) => {
-                    onSearchInputValueChange(event.target.value);
-                  }}
-                />
-                {searchInputValue !== "" && (
+
+                {isNonImportedCustomToken ? (
                   <button
-                    className={classes.clearSearchButton}
-                    onClick={() => onSearchInputValueChange("")}
+                    className={classes.addTokenButton}
+                    onClick={() => {
+                      addWrappedToken({ token })
+                        .then((newToken) => {
+                          onNavigateToTokenAdder(newToken);
+                        })
+                        .catch(() => {
+                          onNavigateToTokenAdder(token);
+                        });
```

Comment:
We should use `callIfMounted` in these `then` and `catch`.

Reply:
Why? f the component unmounts this `onNavigate...` callbacks shouldn't be available.

Reply:
You're right!